rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/ControllerPagination.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.controllers.v3
+
+import za.co.absa.enceladus.rest_api.utils.implicits._
+
+import java.util.Optional
+import scala.util.{Failure, Success, Try}
+
+object ControllerPagination {
+  val DefaultOffset: Int = 0
+  val DefaultLimit: Int = 20
+
+  /**
+   * Offset value is extracted from `optField`, otherwise `defaultValue` returned
+   *
+   * @param optField
+   * @param defaultValue when extraction fails, this value is used. Default = [[ControllerPagination#DefaultOffset()]]
+   * @return
+   */
+  def extractOptionalOffsetOrDefault(optField: Optional[String], defaultValue: Int = DefaultOffset): Int = {
+    extractOptionOffsetOrDefault(optField.toScalaOption, defaultValue)
+  }
+
+  /** Offset value is extracted from `optField`, otherwise `defaultValue` returned
+   *
+   * @param optField
+   * @param defaultValue when extraction fails, this value is used. Default = [[ControllerPagination#DefaultOffset()]]
+   * @return
+   */
+  def extractOptionOffsetOrDefault(optField: Option[String], defaultValue: Int = DefaultOffset): Int = {
```

Comment:
I deliberately wanted to overload that public method so that the pagination helper object (`ControllerPagination`) is versatile enough not to care if `Optional[String]` or `Option[String]` is passed - this way, both works without the caller needing to think about this. (Because here, `Optional` (from Java) is what we need while `Option` (from Scala) is the signature we want there to be - because we are writing primarily Scala code.

Does it make sense? How strongly do you feel about this?

Reply:
It does make sense. Cool keep it as it is.
Cool it's deliberate and thought through 👍